server/src/main/java/org/elasticsearch/ingest/PipelineConfiguration.java

```diff
@@ -114,13 +161,22 @@ public Integer getVersion() {
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject();
         builder.field("id", id);
-        builder.field("config", getConfigAsMap());
+        builder.field("config", config);
         builder.endObject();
         return builder;
     }
 
     public static PipelineConfiguration readFrom(StreamInput in) throws IOException {
-        return new PipelineConfiguration(in.readString(), in.readBytesReference(), in.readEnum(XContentType.class));
+        final String id = in.readString();
+        final Map<String, Object> config;
+        if (in.getTransportVersion().onOrAfter(TransportVersions.INGEST_PIPELINE_CONFIGURATION_AS_MAP)) {
+            config = in.readGenericMap();
+        } else {
+            final BytesReference bytes = in.readBytesReference();
```

Comment:
NIT: Not too important since it's BwC and these things aren't gigantic but in the spirit of doing this consistently, you could use `readSlicedBytesReference` actually here.

Reply:
Nice, okay, 8467326d013fe51de6ba418c9191c4a267a06313.

Reply:
> in the spirit of doing this consistently, you could use `readSlicedBytesReference` actually here.

@original-brownbear can you explain more about this? `readSlicedBytesReference()` just calls `readBytesReference()`, so is it just for naming consistency?

Reply:
That delegation is only the default behavior. For the real world network buffers we have here we have overriding implementations that just slice the underlying buffer without copying.

Reply:
Gotcha, thanks for the explanation!